lib/active_metrics/connection_adapters/abstract_adapter.rb

```diff
@@ -35,6 +35,32 @@ def write(metric)
       def write_multiple(*_metrics)
         raise NotImplementedError, "must implemented by the adapter"
       end
+
+      def transform_parameters(resource, interval_name, _start_time, _end_time, rt_rows)
```

Comment:
@Ladas can this stay in perf_process?  It seems weird to call into ActiveMetrics then call back out to `Metric::Helper.process_derived_columns`

Reply:
right, I am extracting it here since PG adapter will have specific code. Right now we are transforming the data like 4 times from the original data, ending with original data format. :-)